server/src/server.rs

```diff
@@ -425,6 +433,27 @@ impl<B, L> Builder<B, L> {
 		}
 	}
 
+	/// Configure the server to only serve JSON-RPC HTTP requests.
+	///
+	/// Default: both http and ws are enabled.
+	pub fn http_only(mut self) -> Self {
+		self.settings.enable_http = true;
+		self.settings.enable_ws = false;
+		self
+	}
+
+	/// Configure the server should only serve JSON-RPC WebSocket requests.
+	///
+	/// That implies that server just denies HTTP requests which isn't a WebSocket upgrade request
+	/// except if you have enabled middleware which are processed before this check occurs.
+	///
+	/// Default: both http and ws are enabled.
+	pub fn ws_only(mut self) -> Self {
```

Comment:
@lexnv this is a bit opinionated but I would like to have it for consistency.

Reply:
this would disable all JSON-RPC HTTP requests just reply 403 on all HTTP requests (including middleware as well)